Unnecessary pragma directives to disable CA1816

Calling `GC.SuppressFinalize(this)` in `DisposeAsync` does not trigger warning CA1816, so pragma directive to disable it are unnecessary (this applies to 2 code snippets on the page):

```diff
public async ValueTask DisposeAsync()
{
    // Perform async cleanup.
    await DisposeAsyncCore();

    // Dispose of unmanaged resources.
    Dispose(false);

-#pragma warning disable CA1816 // Dispose methods should call SuppressFinalize
    // Suppress finalization.
    GC.SuppressFinalize(this);
-#pragma warning restore CA1816 // Dispose methods should call SuppressFinalize
}
```

I've checked it in .NET 6.0.402, don't have other SDK version installed to verify.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 61a961dc-3ebf-2a96-a6eb-2e0b62ab909f
* Version Independent ID: d72276e1-9059-815f-dc14-a03a7544ab87
* Content: [Implement a DisposeAsync method](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync#implement-both-dispose-and-async-dispose-patterns)
* Content Source: [docs/standard/garbage-collection/implementing-disposeasync.md](https://github.com/dotnet/docs/blob/main/docs/standard/garbage-collection/implementing-disposeasync.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**


---
[Associated WorkItem - 57757](https://dev.azure.com/msft-skilling/Content/_workitems/edit/57757)

--- a/docs/standard/garbage-collection/implementing-disposeasync.md
+++ b/docs/standard/garbage-collection/implementing-disposeasync.md
@@ -3,7 +3,7 @@ title: Implement a DisposeAsync method
 description: Learn how to implement DisposeAsync and DisposeAsyncCore methods to perform asynchronous resource cleanup.
 author: IEvangelist
 ms.author: dapine
-ms.date: 11/30/2022
+ms.date: 01/02/2023
 dev_langs:
   - "csharp"
 helpviewer_keywords:
@@ -49,10 +49,8 @@ public async ValueTask DisposeAsync()
     // Dispose of unmanaged resources.
     Dispose(false);
 
-#pragma warning disable CA1816 // Dispose methods should call SuppressFinalize
     // Suppress finalization.
     GC.SuppressFinalize(this);
-#pragma warning restore CA1816 // Dispose methods should call SuppressFinalize
 }
 ```
 

--- a/docs/standard/garbage-collection/snippets/dispose-async/ExampleConjunctiveDisposable.cs
+++ b/docs/standard/garbage-collection/snippets/dispose-async/ExampleConjunctiveDisposable.cs
@@ -14,9 +14,7 @@
         await DisposeAsyncCore().ConfigureAwait(false);
 
         Dispose(disposing: false);
-#pragma warning disable CA1816 // Dispose methods should call SuppressFinalize
         GC.SuppressFinalize(this);
-#pragma warning restore CA1816 // Dispose methods should call SuppressFinalize
     }
 
     protected virtual void Dispose(bool disposing)

